File: code/server/src/github.com/jason-wilmans/votesComponent/entities/vote.go
package entities

import (
	"github.com/satori/go.uuid"
	"github.com/muroc/prego"
)

type Vote struct {
	Id   uuid.UUID
	Name string
}

func New(name string) Vote {
	precond.True(len(name) > 0, "")

	id, _ := uuid.NewV4()
	return Vote{id, name}
}

func (this *Vote) GetId() uuid.UUID {
	return this.Id
}

func (this *Vote) SetName(name string) {
	precond.NotNil(name, "")
	precond.True(len(name) > 0, "")

	this.Name = name
}

func (this *Vote) GetName() string {
	return this.Name
}

File: code/webClient/src/app/model/vote.ts
import {UUID} from "uuid";
import {Option} from "./option";

export class Vote {
  private readonly Id: string;
  public Name: string;
  private Options: Option[];

  constructor(name: string, id?: string) {

    if (id) {
      this.Id = id;
    } else {

    }

    this.Name = name;
    this.Options = [];
  }

  public GetId(): string {
    return this.Id;
  }

  public SetName(name: string) {
    this.Name = name;
  }

  public GetName(): string {
    return this.Name;
  }

  public Add(option: Option) {
    this.Options.push(option)
  }

  GetOptions(): Option[] {
    const result: Option[] = [];
    this.Options.forEach((option) => {
      result.push(option)
    });
    return result;
  }

}
